lib/vim-state.coffee

```diff
@@ -654,6 +662,16 @@ class VimState
     text = @getRegister(name)?.text
     @editor.insertText(text) if text?
 
+  ensureCursorIsWithinLine: (cursor) =>
+    return if @processing or @mode is 'visual' or @mode is 'insert'
```

Comment:
Some motions, esp. with counts, invoke several Atom motions, each notifies us about cursor movement, and they all can happen in normal mode. We don't want to interfere until that's done. 

But since it needs explanation, it could be written clearer or better; suggestions welcome.